src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -405,6 +423,9 @@ private void BackgroundLogger()
                                     // we don't want to have fsync here, we just
                                     // want to send it to the OS
                                     currentFile.Flush(flushToDisk: false);
+                                    if (_hasEntries.IsSet == false)
```

Comment:
Wondering if we will be able to hit this on a live huge system?

Reply:
What do you mean? We want to avoid this on an ongoing basic, and we are, at this point, going to call the `Wait` method anyway.
Since we are about to go idle, we can afford to check if this is still up to date, no?

Reply:
Ok